apps/meteor/client/hooks/useJoinRoom.ts

```diff
@@ -2,13 +2,17 @@ import type { IRoom } from '@rocket.chat/core-typings';
 import { useMutation, useQueryClient } from '@tanstack/react-query';
 
 import { sdk } from '../../app/utils/client/lib/SDKClient';
+import type { AppPreventedError } from '../lib/errors/AppPreventedError';
+import { dispatchToastMessage } from '../lib/toast';
 
 type UseJoinRoomMutationFunctionProps = {
 	rid: IRoom['_id'];
 	reference: string;
 	type: IRoom['t'];
 };
 
+const isAppError = (error: AppPreventedError | Error): error is AppPreventedError => (error as AppPreventedError).reason !== undefined;
```

Comment:
```suggestion
const isAppError = (error: AppPreventedError | Error): error is AppPreventedError => error instanceof AppPreventedError;
```

Reply:
I didn't test, but doesn't this work? I think a bunch of errors can have the "reason" property.

Reply:
Otherwise, I think you shouldn't create `AppPreventedError`. You could just cast `error: unknown` and test `'reason' in error`. Just throwing ideas

Reply:
In fact, the reason could appear for another errors, I wasn't totally happy or sure with my solution, it looks like your recommendation is really good if it works. I will test it and probably approve...

I was trying to follow some pattern that we discussed before in our frontend team about given errors a proper description and details.